Issue 2020  remove ansicon warning

weedySeaDragon opened the pull request:
I just removed the code that tested for and warned about ANSICON.  
I put documentation (a note about using ANSICON or other utilities) into ```features/configuration/read_options_from_file.feature```.  I'm not sure if that's the right or best place for it.  It seemed to be the most appropriate place.

myronmarston commented on features/configuration/read_options_from_file.feature:
```diff
@@ -16,6 +16,24 @@ Feature: read command line configuration options from files
+      [MinGW](http://www.mingw.org/) (Minimalist GNU for Windows)
+
+  More information can be found by searching for information about those solutions.
+
```
:+1: This is well written.  Thanks for writing this up!

myronmarston replied:
Although, I'm realizing that a file named `read_options_from_file.feature` isn't a place users are likely to look for this info.  WDYT about creating a `features/configuration/color.feature` file with color config info including this?  If you go that route, would be nice to add the file to `features/.nav` so we can put it in the ordering we want.

weedySeaDragon replied:
That makes good sense to me.  That will be much more obvious on [Relish](http://www.relishapp.com/rspec/rspec-core/docs) and anywhere else.  I'll make an entry in ```.nav```.  I'll put it after ```- read_options_from_file.feature``` because that references color and also because if people are having trouble with color, they'll see it quickly.  But of course I'm happy to put it where-ever you think it should go in ```.nav```.

myronmarston commented on lib/rspec/core/configuration.rb:
```diff
@@ -711,15 +711,7 @@ def color_enabled?(output=output_stream)
-          @color = true
-        end
+        @color = !!true_or_false
       end
```
I think that this method can be replaced by a simpler `attr_writer :color`.

weedySeaDragon replied:
Sounds good.  I didn't know if that method needed to hang around or not.  I'll change it to ```attr_writer :color```

weedySeaDragon commented:
made the changes.  If they're ok, let me know if you want me to squash the commits.

myronmarston commented on features/configuration/color.feature:
```diff
@@ -0,0 +1,22 @@
+Feature: Windows may require additional solutions to display color
+
+  The output uses ANSI escapte codes to show text in color.  Windows
```
s/escapte/escape/

myronmarston replied:
Also, do you think it's worth turning "ANSI escape codes" into a link to https://en.wikipedia.org/wiki/ANSI_escape_code ?

myronmarston commented:
Yes, please squash the commits.

Looks like the `attr_writer :color` change we discussed hasn't been made yet.

weedySeaDragon commented:
I added ``attr_writer :color`` Not sure where you wanted it, so I added it right where the method was. (Since other macros are used a lot in the file, I thought doing it that way might be fine.)  Thanks for catching that.  I added the link to Wikipedia's ANSI escape codes, too.  I've seen a lot of people asking about "those weird characters..." on StackOverflow, so I'm sure some will appreciate the information.

commits squashed

myronmarston commented on lib/rspec/core/configuration.rb:
```diff
@@ -695,9 +695,7 @@ def full_backtrace=(true_or_false)
-      def color
-        value_for(:color) { @color }
-      end
+      attr_writer :color
```
This is a replacement for the `def color=` method, not the `def color` method.

weedySeaDragon replied:
Oh how embarrassing. (:head_smack:) Fixed.

myronmarston commented:
Thanks, @weedySeaDragon!
